### PR TITLE
[backbone-router] add build options for backbone router features

### PR DESCRIPTION
**Pull request by simonlingoogle**
Depends on: https://github.com/openthread/openthread/pull/5955

### CODE REVIEW

File: src/backbone_router/backbone_agent.cpp [hunk 1]
```diff
@@ -203,3 +205,5 @@ void BackboneAgent::HandleBackboneRouterNdProxyEvent(otBackboneRouterNdProxyEven
 
 } // namespace BackboneRouter
 } // namespace otbr
+
+#endif
```
Comment (Moandor-y) [inline src/backbone_router/backbone_agent.cpp:209]:
```suggestion
#endif // OTBR_ENABLE_BACKBONE_ROUTER_DUA_NDPROXYING
```

File: src/backbone_router/backbone_agent.hpp [hunk 2]
```diff
@@ -119,11 +121,13 @@ class BackboneAgent
...
 /**
  * @}
  */
 
 } // namespace BackboneRouter
 } // namespace otbr
+
+#endif // OTBR_ENABLE_BACKBONE_ROUTER_DUA_NDPROXYING
+
+#endif
```
Comment (Moandor-y) [inline src/backbone_router/backbone_agent.hpp:133]:
```suggestion
#endif // BACKBONE_ROUTER_BACKBONE_AGENT_HPP_
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -96,11 +96,26 @@ if (OTBR_BACKBONE_ROUTER)
...
 endif()
 
+
+option(OTBR_BACKBONE_ROUTER_DUA_NDPROXYING "Enable Backbone Router DUA ND Proxying" OFF)
+
+if (OTBR_BACKBONE_ROUTER_DUA_NDPROXYING)
+    target_compile_definitions(otbr-config INTERFACE
+            OTBR_ENABLE_BACKBONE_ROUTER_DUA_NDPROXYING=1
+    )
+    set(OT_BACKBONE_ROUTER_DUA_NDPROXYING ON CACHE BOOL "Disable Backbone Router DUA ND Proxying in OpenThread" FORCE)
```
Comment (Moandor-y) [inline CMakeLists.txt:112]:
```suggestion
    set(OT_BACKBONE_ROUTER_DUA_NDPROXYING OFF CACHE BOOL "Disable Backbone Router DUA ND Proxying in OpenThread" FORCE)
```

Looks like a typo?

Comment (simonlingoogle) [reply CMakeLists.txt:112]: It's not. If `OTBR_BACKBONE_ROUTER_DUA_NDPROXYING` is ON, we are tuning `OT_BACKBONE_ROUTER_DUA_NDPROXYING` too by **FORCE**.

Comment (Moandor-y) [reply CMakeLists.txt:112]: But why does the description say "disable"?

Comment (simonlingoogle) [reply CMakeLists.txt]: fixed

**Review by Moandor-y — APPROVED** [review]
LGTM :+1:

**Review by wgtdkp — APPROVED** [review]

**Comment by simonlingoogle** [comment]
@librasungirl rebased on master, please take a look.

File: CMakeLists.txt [hunk 4]
```diff
@@ -96,11 +96,26 @@ if (OTBR_BACKBONE_ROUTER)
...
             OTBR_ENABLE_BACKBONE_ROUTER=1
     )
+    set(OTBR_BACKBONE_ROUTER_DUA_NDPROXYING ON CACHE BOOL "Backbone Router enables DUA ND Proxying by default")
     set(OT_THREAD_VERSION 1.2 CACHE STRING "Backbone Router requires Thread 1.2 or higher" FORCE)
     set(OT_BACKBONE_ROUTER ON CACHE BOOL "Enable Backbone Router feature in OpenThread" FORCE)
     set(OT_SERVICE ON CACHE BOOL "Backbone Router requires Thread network service" FORCE)
 endif()
 
+
+option(OTBR_BACKBONE_ROUTER_DUA_NDPROXYING "Enable Backbone Router DUA ND Proxying" OFF)
```
Comment (librasungirl) [inline CMakeLists.txt:106]: It seems the new `OTBR_BACKBONE_ROUTER_DUA_NDPROXYING` is equivalent with `OTBR_ENABLE_BACKBONE_ROUTER` from usage's perspective. May keep using  `OTBR_ENABLE_BACKBONE_ROUTER`?

Comment (simonlingoogle) [reply CMakeLists.txt:106]: Agree that we can keep only `OTBR_ENABLE_BACKBONE_ROUTER`.

**Review by librasungirl — APPROVED** [review]

File: CMakeLists.txt (hunk 4 above)
Comment (simonlingoogle) [reply CMakeLists.txt]: fixed

File: CMakeLists.txt [hunk 5]
```diff
@@ -96,6 +96,7 @@ if (OTBR_BACKBONE_ROUTER)
     target_compile_definitions(otbr-config INTERFACE
             OTBR_ENABLE_BACKBONE_ROUTER=1
     )
+    set(OT_BACKBONE_ROUTER_DUA_NDPROXYING ON CACHE BOOL "Enable Backbone Router DUA ND Proxying in OpenThread" FORCE)
```
Comment (librasungirl) [inline CMakeLists.txt:99]: Could be removed?
Below OT_BACKBONE_ROUTER is passed which would set default value.

```
option(OT_BACKBONE_ROUTER "enable backbone router functionality")
if(OT_BACKBONE_ROUTER)
    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE=1")
    set(OT_BACKBONE_ROUTER_DUA_NDPROXYING ON CACHE BOOL "Enable DUA NDProxying by default")
    set(OT_BACKBONE_ROUTER_MULTICAST_ROUTING ON CACHE BOOL "Enable Multicast Routing by default")
endif()
```

**Review by librasungirl — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**